server/src/main/java/com/linecorp/centraldogma/server/internal/api/ProjectServiceV1.java

```diff
@@ -189,10 +189,11 @@ public CompletableFuture<Void> purgeProject(@Param String projectName, Author au
     @Consumes("application/json-patch+json")
     @Patch("/projects/{projectName}")
     @RequiresAdministrator
-    public CompletableFuture<ProjectDto> patchProject(@Param String projectName, JsonNode node, Author author) {
+    public CompletableFuture<ProjectDto> patchProject(@Param String projectName, JsonNode node, Author author,
```

Comment:
Question) This method updates the metadata and immediately returns the project via `projectApiManager.getProject`.

However, `projectApiManager.getProject` relies on `Project#metadata` which is updated asynchronously.
Were you able to confirm this API will work correctly in a distributed setting? It seems like if a thread reschedule is done in the following line the most recent metadata may not be accessed.

https://github.com/line/centraldogma/blob/4d4af7e350226986d1e3dc2ee3566e44b6f48428/server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/GitRepository.java#L1525

Reply:
Good point. I'll improve `RepositoryListener` to be notified synchronously while a commit is performed in the F/U work.